StorageUtility/Windows/DeviceHandle.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#include "Device.h"
+#include "DeviceHandle.h"
+#include "StorageUtility.h"
+
+namespace vtStor
+{
+
+cDeviceHandle::cDeviceHandle(std::shared_ptr<cDeviceInterface> Device)
+{
+    String devicePath;
+    GetDevicePath(devicePath, Device);
+    GetStorageDeviceHandle(devicePath, m_DeviceHandle);
+    //TODO: handle errors
+}
+
+cDeviceHandle::~cDeviceHandle()
+{
+    CloseDeviceHandle(m_DeviceHandle);
```

Comment:
The user in this case is the cDrive object. Once the handle is given, the handle can be close from outside the cDeviceHandle object. This is a problem if the cDeviceHandle also have a reference to the device handle. Therefore, I made a design decision that the cDeviceHandle object should not store any device handles. Instead, it should give out a new one every time.